### PR TITLE
fix: issue with isModelGenerating when switching between multiple models

### PR DESCRIPTION
**Pull request by NorbertKlockiewicz**
## Description
There was a problem when user was switching between multiple llms in one component, to fix this issue I removed code related to handling strict mode problems(strict mode was causing problems with event listeners also so I think we should ignore it as it isn't the best with background tasks), I've also remove deleteModule native function as it wasn't really doing anything, now user can download multiple llms within one component and seamlessly switch between them without bugs.
The problem was mentioned in issue #42 

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improves or adds clarity to existing documentation)

### Tested on
- [x] iOS
- [x] Android

### Checklist
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly
- [ ] My changes generate no new warnings

### CODE REVIEW

File: android/src/main/java/com/swmansion/rnexecutorch/LLM.kt [hunk 1]
```diff
@@ -74,8 +75,8 @@ class LLM(reactContext: ReactApplicationContext) :
     contextWindowLength: Double,
     promise: Promise
   ) {
-    if (llamaModule != null || isFetching) {
-      promise.reject("Model already loaded", "Model is already loaded or fetching")
+    if (isFetching) {
+      promise.reject("Model already loaded", "Model is already fetching")
```
Comment (chmjkb) [inline android/src/main/java/com/swmansion/rnexecutorch/LLM.kt]: The first argument seems wrong

File: ios/RnExecutorch/LLM.mm [hunk 2]
```diff
@@ -54,8 +54,8 @@ - (void)updateDownloadProgress:(NSNumber *)progress {
 - (void)loadLLM:(NSString *)modelSource tokenizerSource:(NSString *)tokenizerSource systemPrompt:(NSString *)systemPrompt contextWindowLength:(double)contextWindowLength resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject {
   NSURL *modelURL = [NSURL URLWithString:modelSource];
   NSURL *tokenizerURL = [NSURL URLWithString:tokenizerSource];
-    
-  if(self->runner || isFetching){
+  
+  if(isFetching){
     reject(@"model_already_loaded", @"Model and tokenizer already loaded", nil);
```
Comment (chmjkb) [inline ios/RnExecutorch/LLM.mm]: Is this error message correct? you're not checking if its loaded

File: android/src/main/java/com/swmansion/rnexecutorch/LLM.kt [hunk 3]
```diff
@@ -148,10 +149,6 @@ class LLM(reactContext: ReactApplicationContext) :
     llamaModule!!.stop()
   }
 
-  override fun deleteModule() {
```
Comment (chmjkb) [inline android/src/main/java/com/swmansion/rnexecutorch/LLM.kt]: Why are we getting rid of this?

Comment (NorbertKlockiewicz) [reply android/src/main/java/com/swmansion/rnexecutorch/LLM.kt]: I noticed that it doesn't really change memory usage, but let's leave it for now as it's not breaking anything

**Review by chmjkb — APPROVED** [review]

**Pull request merged**